flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/PartitionRequestQueue.java

```diff
@@ -354,4 +404,64 @@ public void operationComplete(ChannelFuture future) throws Exception {
 			}
 		}
 	}
+
+	private static class RegisterPeriodicFlushEvent {
+		private final NetworkSequenceViewReader reader;
+		private final long flushTimeout;
+
+		public RegisterPeriodicFlushEvent(NetworkSequenceViewReader reader, long flushTimeout) {
+			this.reader = checkNotNull(reader);
+			this.flushTimeout = flushTimeout;
+		}
+
+		public NetworkSequenceViewReader getReader() {
+			return reader;
+		}
+
+		public long getFlushTimeout() {
+			return flushTimeout;
+		}
+	}
+
+	private static class PeriodicFlushes {
```

Comment:
We should clear the related maps or cancel the future in `releaseAllResources` caused by `exceptionCaught` or `channelInactive`?